org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/launch/BaseLaunchRequestJob.java

```diff
@@ -41,21 +43,32 @@
 import org.eclipse.buildship.core.util.collections.CollectionsUtils;
 import org.eclipse.buildship.core.util.file.FileUtils;
 import org.eclipse.buildship.core.util.gradle.GradleDistributionFormatter;
+import org.eclipse.buildship.core.util.gradle.GradleDistributionWrapper;
 import org.eclipse.buildship.core.util.progress.DelegatingProgressListener;
 import org.eclipse.buildship.core.util.progress.ToolingApiJob;
 import org.eclipse.buildship.core.workspace.ModelProvider;
 
 /**
- * Base class to execute {@link SingleBuildRequest} instances in job.
+ * Base class to execute Gradle builds in a job.
  */
 public abstract class BaseLaunchRequestJob extends ToolingApiJob {
 
+    /**
+     * Common interface to retrieve and execute build and test launch requests. This interface is
+     * needed since there's no common {@code run()} method defined on the
+     * {@link LongRunningOperation} subclasses.
+     */
+    public static interface Launcher {
```

Comment:
I've adjusted the implementation but I think there's still room for improvement.